src/utils/restClient/index.ts

```diff
@@ -20,7 +20,7 @@ class RestClient {
 		const searchParams = new URLSearchParams();
 
 		for (const key in queryParams) {
-			if (Object.prototype.hasOwnProperty.call(queryParams, key)) {
+			if (typeof queryParams?.[key] !== 'undefined') {
```

Comment:
Thank you for your willingness to contribute!
An analysis of the proposed solution was carried out, taking into account:

This more robust and secure implementation avoids surprises with legacy data and undefined values, ensuring that our application only processes the necessary and anticipated information.

By combining both checks, the function becomes more secure and flexible when dealing with different data sources, especially if that data is derived from user input or external APIs.

```suggestion
      if (Object.prototype.hasOwnProperty.call(queryParams, key) && typeof queryParams[key] !== 'undefined') {
```

Reply:
Thank you, your suggestion combining both validations will be taken into account.

Reply:
I can help make your application more robust and lightweight if needed.

From small fixes such as improvements in typing security and replacement of export defaults only by export:
![image](https://github.com/user-attachments/assets/9072b7b4-2d05-4f5e-b400-03e3d31a79d4)
Even more relevant changes like removing node-fetch and prioritizing the use of the native https module (saving 481kB), or using native crypto module instead uuid(need discussion, node:crypto depends of the enviroment to randomize but saving 120kB).

Reply:
Thank you for your willingness to contribute!

But we ask that you open a new Pull Request using the pre-defined models and including as much information as possible about the improvements mentioned above, thus making the analysis process faster and more effective.